Snowflake.Data/Core/SFResultSetMetaData.cs

```diff
@@ -37,24 +37,20 @@ class SFResultSetMetaData
         /// </summary>
         private Dictionary<string, int> columnNameToIndexCache = new Dictionary<string, int>();
 
-        internal SFResultSetMetaData(QueryExecResponseData queryExecResponseData)
+        internal SFResultSetMetaData(QueryExecResponseData queryExecResponseData, SFSession session)
         {
             rowTypes = queryExecResponseData.rowType;
             columnCount = rowTypes.Count;
             statementType = findStatementTypeById(queryExecResponseData.statementTypeId);
             columnTypes = InitColumnTypes();
-            
-            foreach (NameValueParameter parameter in queryExecResponseData.parameters)
+
+            if (session.ParameterMap.ContainsKey(SFSessionParameter.DATE_OUTPUT_FORMAT))

```

Comment:
Session should always have these parameters as they are returned with the first response, am I right?

Reply:
Yes but for some unit test cases the session is made from a mocked login response which doesn't have these parameters.